chainer/training/updater.py

```diff
@@ -10,7 +10,19 @@ class Updater(object):
 
     """Interface of updater objects for trainers.
 
-    TODO(beam2d): document it.
+    :class:`~chainer.training.Updater` implements a training iteration
+    as :meth:`update`.
+
+    Typically, the updating iteration proceeds as follows.
+
+    - Fetch a minibatch from :class:`~chainer.dataset.DatasetMixin`.
```

Comment:
Also as we usually retrieve minibatches via iterators (as `StandardUpdater` does), it could be better to add "via iterators" in the end of the sentence.